Draft-Accepted/Install-PowerShell.md

```diff
@@ -0,0 +1,217 @@
+automation solution available to many of them. While customers are missing out on useful new
+features that they may wish to have, they are certainly not receiving the performance and security
+improvements that come with each new release.
+
+This specification is an effort to help IT admins that wish to use new and updated releases of
+PowerShell on Windows and to easily install PowerShell from **Windows PowerShell 5.1** with a
+cmdlet. While this may not help customers that have post-deployment policy restrictions, it will
+help the majority of customers discover and update to the latest version of PowerShell for Windows.
+
+```
+As an admin,
+I can use Windows PowerShell 5.1 to provision multiple hosts with PowerShell 7 over PSRemoting,
+so that I can leverage new capabilities in PS7.
+```
+
+```
+As a user,
+I can discover and install PowerShell 7 from Windows PowerShell 5.1 on my Windows machine,
+so that I can leverage new capabilities in PS7.
+```
+
+## Shipping in Windows
+
+There are several issues that prevent shipping PowerShell 7 in Windows;
+
+- Size
+- Support lifecycle differences
+
+### Size constraints
+
+ A complete installation of PowerShell 7 currently is around 200 MB (unpacked). This is due to
+ including everything (modules and .NET assemblies) to provide parity with what Windows PowerShell
+ 5.1 with .NET Framework provides. There is currently no plan by Windows or the .NET team to ship
+ newer .NET in Windows, so PS7 will need to continue to ship .NET with PS7.
+
+### Support lifecycle
+
+ Windows has a 5+5 year support lifecycle. .NET supports up to 3 years for Long Term Support (LTS)
+ versions. So even if size was not an issue, we cannot check in a version of PS7 with a .NET that is
+ not supported while Windows itself is supported.
+
+### Solution
+
+To solve these issues, the proposal is to ship a new cmdlet in Windows and the PowerShell Gallery
+that would;
+
+- Be discoverable using normal PowerShell conventions
+- Perform a simple download and install of the latest stable release of PowerShell
+- Provide advanced options with parameters;
+  - Specific version
+  - Stable, Preview or LTS
+  - Scope - location of installation based on permissions
+
+## Goals/Non-Goals
+
+Goals:
+
+- Be discoverable as other built-in cmdlets
+- By default, download and install the latest stable release to the users Scope
+  - AllUsers - if user has administrative privileges
+  - CurrentUser - if the user does not have administrative privileges
+- Allow users to use remoting for install at scale with Invoke-Command
+- Allow advanced users to specify the version of PowerShell
+- Allow advanced users to specify scope (location of installation)
+- Allow advanced users to specify channel Stable/Preview/LTS
+- Users should receive a warning to continue the installation (ConfirmImpact=High)
+
+Non-goals:
+
+- Removal of PowerShell (Remove-PowerShell7)
+- Updating of PowerShell (Update-PowerShell7)
```

Comment:
I think updating PowerShell should be included. 
Specifically. If a user runs the new command today (7.2.4 is current) and in few months opens Windows PowerShell and runs it again, what happens ? If it doesn't update to  7.2.5 or 7.3 or 8 they need to be given a very good reason why not. Ditto if they run the command from 7.  The version installed on a machine depending on when the command was _first_ run, not when it was _last_ run would be a bad thing all round.

Reply:
Thank you - for this version, updating and rollback is outside the scope.